### PR TITLE
fix last_viewed sorting (doesn't update as users visit libraries)

### PR DESCRIPTION
**Pull request by ahsu1230**
Last Viewed sorting doesn't dynamically resort whenever you visit your libraries. You'd have to refresh in order to get the correct ordering. This uses angular's `Date.now()` to dynamically update the nav libs list.

@andrewconner @yipingliao

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/layout/nav/nav.js [hunk 1]
```diff
@@ -98,6 +98,19 @@ angular.module('kifi')
...
         scope.$on('$destroy', deregisterChangedLibrary);
 
+        var deregisterLibraryVisited = $rootScope.$on('lastViewedLib', function (e, lib) {
+          if (lib.name) {
+            var ind = _.findIndex(scope.userLibsToShow, function(l) { return l.name === lib.name; });
+            if (scope.userLibsToShow[ind]) {
+              scope.userLibsToShow[ind].lastViewed = Date.now();
+            }
+            if (scope.sortingMenu.option === 'last_viewed') {
+              updateNavLibs();
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/layout/nav/nav.js:108]: only update the list if you're on `last_viewed` sorting option. But, still record times of which libraries you're visiting, so correct sorting persists whether you're in `last_viewed` sorting or not

File: kifi-backend/modules/shoebox/angular/src/layout/nav/nav.js [hunk 2]
```diff
@@ -98,6 +98,19 @@ angular.module('kifi')
         });
         scope.$on('$destroy', deregisterChangedLibrary);
 
+        var deregisterLibraryVisited = $rootScope.$on('lastViewedLib', function (e, lib) {
+          if (lib.name) {
+            var ind = _.findIndex(scope.userLibsToShow, function(l) { return l.name === lib.name; });
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/layout/nav/nav.js]: `var index = _.findIndex(scope.userLibsToShow, { 'name': lib.name })`
If you don't like `index` try `idx` (more idiomatic than `ind`?)

**Comment by yipingliao** [comment]
Nice!

File: kifi-backend/modules/shoebox/angular/src/layout/nav/nav.js (hunk 2 above)
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/layout/nav/nav.js]: :+1: dohhh I keep forgetting that notation. Thanks

**Pull request merged**